sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java

```diff
@@ -139,6 +146,11 @@ public void finalizeCheckpoint() throws IOException {
         // nothing to do
       }
     }
+
+    /* Get offset limit for unbounded source split checkpoint. */
+    default byte[] getOffsetLimit() throws IOException {
```

Comment:
maybe we should just throw a runtime exception? it seems like this shouldn't have to do IO and IOException needs to be handled at callers if method is annoated as throwing non-runtime exception

Reply:
Ah I see the difference now, updated to RuntimeException.